pyk4a/pyk4a.py

```diff
@@ -14,81 +15,99 @@
     from typing import TypedDict
 
 
-# k4a_wait_result_t
-class Result(Enum):
-    Success = 0
-    Failed = 1
-    Timeout = 2
-
-
-class K4AException(Exception):
-    pass
-
-
-class K4ATimeoutException(K4AException):
-    pass
-
-
 class PyK4A:
     TIMEOUT_WAIT_INFINITE = -1
 
     def __init__(self, config: Optional[Config] = None, device_id: int = 0, thread_safe: bool = True):
         self._device_id = device_id
         self._config: Config = config if (config is not None) else Config()
         self.thread_safe = thread_safe
+        self._device_handle: Optional[object] = None
+        self._calibration: Optional[Calibration] = None
         self.is_running = False
 
     def __del__(self):
         if self.is_running:
             self.disconnect()
+        elif self.opened:
+            self.close()
 
-    def connect(self):
+    @property
+    def opened(self) -> bool:
+        return self._device_handle is not None
+
+    def open(self):
+        """
+        Open device
+        You must open device before querying any information
+        """
+        if self.opened:
+            raise K4AException("Device already opened")
         self._device_open()
+
+    def close(self):
+        """
+        Close device
+        """
+        self._validate_is_opened()
+        self._device_close()
+
+    def connect(self):
+        """
+        Open device if device not opened, then start cameras and IMU
+        All-in-one function
+        :return:
+        """
+        if not self.opened:
+            self.open()
         self._start_cameras()
+        self._start_imu()
         self.is_running = True
 
     def disconnect(self):
+        """
+        Stop cameras, IMU, ... and close device
+        :return:
+        """
         self._stop_imu()
         self._stop_cameras()
         self._device_close()
         self.is_running = False
```

Comment:
Do you think it is easier to understand if we rename `connect` by `start`?
and `disconnect` by `stop`

Reply:
Also I would put this `start` function at the top of the class, for new users.

Reply:
Renamed and moved up